lib/surrealist.rb

```diff
@@ -129,17 +130,18 @@ def build_schema(instance:, **args)
 
       parameters = config ? config.merge(args) : args
       carrier = Surrealist::Carrier.call(parameters)
-      normalized_schema = Surrealist::Copier.deep_copy(schema, carrier, instance.class.name)
-      hash = Builder.new(carrier, normalized_schema, instance).call
-      carrier.camelize ? Surrealist::HashUtils.camelize_hash(hash) : hash
+      copied_schema = Surrealist::Copier.deep_copy(schema)
+      built_schema = Builder.new(carrier, copied_schema, instance).call
+      wrapped_schema = Surrealist::Wrapper.wrap(built_schema, carrier, instance.class.name)
```

Comment:
This code makes me a little anxious, maybe we should encapsulate this via something like a command pattern here?

Reply:
Yes, you are right, and it would be very nice of you to refactor this part a bit.